README.md

```diff
@@ -8,22 +8,845 @@
+- [`CreateX`](./src/CreateX.sol) should be maximally secure, i.e. no hidden footguns.
+- [`CreateX`](./src/CreateX.sol) should be trustless.
+- [`CreateX`](./src/CreateX.sol) should be stateless.
+- [`CreateX`](./src/CreateX.sol) should be extensible.
+
+The following consequences result from these principles:
+
+- We only use inline assembly if it is required or if the code section itself is based on short and/or audited code.
+- We document the contract to the smallest detail.
+- We extensively fuzz test all functions.
+- We deliberately do not implement special functions for [clones with immutable arguments](https://github.com/wighawag/clones-with-immutable-args), as there is neither a finalised standard nor a properly audited contract version.
+- We do not implement any special functions for [EIP-5202](https://eips.ethereum.org/EIPS/eip-5202) (a.k.a. blueprint contracts), as all existing functions in [`CreateX`](./src/CreateX.sol) are already cost-effective alternatives in our opinion.
+
+## Security Considerations
+
+> **Warning**<br>
+> This contract is unaudited!
+
+Generally, for security issues, see our [Security Policy](./SECURITY.md). Furthermore, you must be aware of the following aspects:
+
+- Several functions allow for reentrancy, however we refrain from adding a mutex lock to keep it as use-case agnostic as possible. Please ensure at the protocol level that potentially malicious reentrant calls do not affect your smart contract system.
+- In the functions [`deployCreate3(bytes32,bytes)`](https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol#L616-L646), [`deployCreate3AndInit(bytes32,bytes,bytes,tuple(uint256,uint256))`](https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol#L727-L762), and [`deployCreate3AndInit(bytes32,bytes,bytes,tuple(uint256,uint256),address)`](https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol#L667-L725) we strongly recommend implementing a permissioned deploy protection by setting the first 20 bytes equal to `msg.sender` in the `salt` to prevent maliciously intended frontrun proxy deployments on other chains.
+- The functions [`deployCreate3(bytes)`](https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol#L648-L665), [`deployCreate3AndInit(bytes,bytes,tuple(uint256,uint256))`](https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol#L801-L834), and [`deployCreate3AndInit(bytes,bytes,tuple(uint256,uint256),address)`](https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol#L764-L799) do not implement any permissioned deploy protection, thus anyone can frontrun the same proxy deployment on other chains. Use with caution!
+
+## Tests
+
+For all (fuzz) tests available in the [`test`](./test) directory, we have consistently applied the [Branching Tree Technique](https://twitter.com/PaulRBerg/status/1682346315806539776). This means that each test file is accompanied by a `.tree` file that defines all the necessary branches to be tested.
```

Comment:
```suggestion
For all (fuzz) tests available in the [`test`](./test) directory, we have consistently applied the [Branching Tree Technique](https://twitter.com/PaulRBerg/status/1682346315806539776) with [bulloak](https://github.com/alexfertel/bulloak). This means that each test file is accompanied by a `.tree` file that defines all the necessary branches to be tested.
```

Reply:
Fixed via https://github.com/pcaversaccio/createx/pull/26/commits/8a1783bcf2719c3d0fb0d5bf0d14200729c323a1. Wanted to use code highlighting for `bulloak` & made you a co-author of the commit of course.